ts/features/bonus/cgn/components/merchants/CgnMerchantListItem.tsx

```diff
@@ -26,6 +24,62 @@ const styles = StyleSheet.create({
     paddingVertical: 16
   }
 });
+const renderCategoryElement = (category: ProductCategory) =>
+  getCategorySpecs(category).fold(undefined, c => (
+    <View style={styles.row}>
+      {c.icon({ height: 22, width: 22, fill: IOColors.bluegrey })}
+      <View hspacer small />
+      <H5 weight={"SemiBold"} color={"bluegrey"}>
+        {I18n.t(c.nameKey).toLocaleUpperCase()}
+      </H5>
+    </View>
+  ));
+
+// This is used to render the product categories of a merchant
+// if there's more than 2 categories available we render a placeholder text to enumerate how many categories
+// are missing from the list
+const CategoriesRow = ({ categories }: Pick<Props, "categories">) => (
+  <>
+    {categories.length > 2
+      ? index(0, [...categories]).fold(undefined, c => (
+          <>
+            {renderCategoryElement(c)}
+            <View
+              hspacer
+              small
+              style={{
+                borderRightColor: IOColors.bluegrey,
+                borderRightWidth: 1
+              }}
+            />
+            <View hspacer small />
+            <H5 color={"bluegrey"} weight={"SemiBold"}>
+              {I18n.t("bonus.cgn.merchantDetail.categories.counting", {
+                count: categories.length - 1
+              }).toLocaleUpperCase()}
+            </H5>
+          </>
+        ))
+      : categories.map(category => (
```

Comment:
If the category name is too long the second category name is not visible:
<img src="https://user-images.githubusercontent.com/11773070/135588471-1af1c9df-4ade-4d4f-aec1-c7d8eecf1022.png" width=300>